### DISCUSSION TITLE
v24.5.27-Database

### DISCUSSION THREAD

**Discussion by RaBa64** [RaBa64/OpenDTU#27] [Announcements]
Merged original OpenDTU changes.
Has somebody checked the database export and import functionality?

<hr /><em>This discussion was created from the release <a href='https://github.com/RaBa64/OpenDTU-Database/releases/tag/v24.5.27-Database'>v24.5.27-Database</a>.</em>

**Comment by nonflammable** [comment]
![obraz](https://github.com/RaBa64/OpenDTU-Database/assets/6041490/9b94fb7b-159f-4dff-a1e2-3a90862b9028)

**Comment by MartinNK** [comment]
Database save/restore worked for me in previous version. Merge some files of database by hand and later with binary append / hexedit. I will test this version today evening or tomorrow. Any news about a replacement of goggle charts? Thanks for your work. Martin

**Reply by RaBa64** [reply]
Thank your for testing the import / export functionality. 

I still have not found a good replacement for the column chart. The calendar chart is using a FOSS since some month now

**Reply by MartinNK** [reply]
Did you prefer a alternative for google charts at the moment or don't you have any other found which fits for the database version needs?

**Reply by RaBa64** [reply]
The Vue column charts I have found, don't fit into the flash memory.

**Comment by nonflammable** [comment]
How can I repair database ?
![obraz](https://github.com/RaBa64/OpenDTU-Database/assets/6041490/ab41cc8e-408d-44aa-a12d-6d407f17fbd6)
![obraz](https://github.com/RaBa64/OpenDTU-Database/assets/6041490/3a7d4d86-ffac-4e09-a936-52c97cb88455)
![obraz](https://github.com/RaBa64/OpenDTU-Database/assets/6041490/091e3626-d4ea-446f-b171-1a311fee22c4)
![obraz](https://github.com/RaBa64/OpenDTU-Database/assets/6041490/8f1457ca-62ac-4265-8652-5ef9d8a92535)

**Reply by RaBa64** [reply]
In the hex editor just copy a valid energy value over the invalid. An import the file back to OpenDTU-Database.

**Reply by nonflammable** [reply]
corrected

![obraz](https://github.com/RaBa64/OpenDTU-Database/assets/6041490/827bce7c-a3f7-4e26-860c-5103361e4d50)

import works fine, but when loading bin file the explorer suggests json file to open
